src/main/kotlin/net/irisshaders/lilybot/extensions/util/ImageChannel.kt

```diff
@@ -0,0 +1,159 @@
+import net.irisshaders.lilybot.utils.DatabaseHelper.getImageChannels
+import net.irisshaders.lilybot.utils.DatabaseHelper.setImageChannel
+import net.irisshaders.lilybot.utils.configPresent
+
+/**
+ * The class the holds the systems that allow a guild to set a channel as image only.
+ *
+ * @since 3.3.0
+ */
+class ImageChannel : Extension() {
+	override val name = "imagechannel"
+
+	override suspend fun setup() {
+		/**
+		 * Image channel commands.
+		 * @author NoComment1105
+		 * @since 3.3.0
+		 */
+		ephemeralSlashCommand {
+			name = "image-channel"
+			description = "The parent command for image channel setting"
+
+			/**
+			 * The command that sets the image channel.
+			 */
+			ephemeralSubCommand {
+				name = "set"
+				description = "Set a channel as an image channel"
+
+				check {
+					anyGuild()
+					configPresent()
+					hasPermission(Permission.ManageGuild)
+				}
+
+				action {
+					setImageChannel(guild!!.id, channel.asChannel().id)
+
+					respond {
+						content = "Set channel as image only"
+					}
+				}
+			}
+
+			/**
+			 * The command that unsets the image channel.
+			 */
+			ephemeralSubCommand {
+				name = "unset"
+				description = "Unset a channel as an image channel"
+
+				check {
+					anyGuild()
+					configPresent()
+					hasPermission(Permission.ManageGuild)
+				}
+
+				action {
+					deleteImageChannel(guild!!.id, channel.asChannel().id)
+
+					respond {
+						content = "Unset channel as image only"
+					}
+				}
+			}
+
+			/**
+			 * The command that returns a list of all image channels for a particular guild.
+			 */
+			ephemeralSubCommand {
+				name = "list"
+				description = "List all image channels in the guild"
+
+				check {
+					anyGuild()
+				}
+
+				action {
+					val imageChannels = getImageChannels(guild!!.id)
+					var channels = ""
+
+					imageChannels.forEach {
+						channels += "<#${it.channelId}> "
+					}
+
+					respond {
+						embed {
+							title = "Image channels"
+							description = "Here are the image only channels in this guild."
+							field {
+								name = "Channels:"
+								value = if (channels != "") channels.replace(" ", "\n") else "No channels found!"
+							}
+						}
+					}
+				}
+			}
+		}
+
+		/**
+		 * The event for checking a channel.
+		 * @since 3.3.0
+		 */
+		event<MessageCreateEvent> {
+			check {
+				anyGuild()
+				failIf { event.message.author.isNullOrBot() } // Fail if null in case an ephemeral message is created
+			}
+
+			action {
+				val imageChannels = getImageChannels(guildFor(event)!!.id)
+
+				for (i in imageChannels) {
+					// If there are no attachments to the message and the channel we're in is an image channel
+					if (event.message.channelId == i.channelId && event.message.attachments.isEmpty()) {
+						// We delay to give the message a chance to populate with an embed, if it is a link to imgur etc.
+						delay(0.1.seconds.millisecondsLong)
```

Comment:
I think this should be a touch longer, I'm able to make it delete some things that should have images